include/execution.hpp

```diff
@@ -436,29 +464,29 @@ namespace std::execution {
           // for the receiver to destroy the coroutine.
           auto fn = [&]<bool Nothrow>(bool_constant<Nothrow>, auto&&... as) noexcept {
             return [&]() noexcept(Nothrow) -> void {
-              set_value((R&&) r, (add_rvalue_reference_t<__await_result_t<A>>) as...);
+              set_value((R&&) r, (add_rvalue_reference_t<__await_result_t<A, __promise_t<R>>>) as...);
             };
           };
-          if constexpr (is_void_v<__await_result_t<A>>)
+          if constexpr (is_void_v<__await_result_t<A, __promise_t<R>>>)
             co_yield (co_await (A &&) a, fn(__nothrow_<R>{}));
           else
-            co_yield fn(__nothrow_<R, __await_result_t<A>>{}, co_await (A &&) a);
+            co_yield fn(__nothrow_<R, __await_result_t<A, __promise_t<R>>>{}, co_await (A &&) a);
         } catch (...) {
           ex = current_exception();
         }
         co_yield [&]() noexcept -> void {
           set_error((R&&) r, (exception_ptr&&) ex);
         };
       }
-    public:
-      template <__awaitable A, receiver R>
-        requires receiver_of<R, __await_result_t<A>>
-      __impl::__op<__id_t<remove_cvref_t<R>>> operator()(A&& a, R&& r) const {
+     public:
+      template <receiver R, __awaitable<__promise_t<R>> A>
+        requires receiver_of<R, __await_result_t<A, __promise_t<R>>>
```

Comment:
Does this requires clause need a `!is_void_v<__await_result_t<A, __promise_t<R>>>` condition to avoid trying to test `receiver_of<void>`?

Reply:
On a conforming compiler, it shouldn't matter. `receiver_of<R, void>` should fail to be satisfied, not cause a hard error.